### ISSUE TITLE
Muon's Newton-Schulz uses fixed 2024 coefficients; per-step optimal ones (Polar Express) orthogonalize more accurately for the same 5 steps

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#375]
**Blocked by #287.** This only matters if Muon becomes the base-run optimizer, and #287 decides whether Muon's #26 win survives a tuned AdamW. If it doesn't, close this `wont-fix: Muon not adopted`.

**Lane:** partial-cpu (`cpu` + `gpu`). **Type:** ideas (recipe).

## What we run

`trm/train/optimizers.py` → `optax.contrib.scale_by_muon` with its defaults: 5 Newton-Schulz steps, all using one fixed quintic `(3.4445, -4.7750, 2.0315)`. Those are Keller Jordan's 2024 coefficients. They were tuned for speed, deliberately without converging: they push the update's singular values into a band of roughly 0.7–1.2 rather than onto 1. So Muon's "orthogonalized" update is only approximately orthogonal, and the approximation error is a fixed, unmeasured property of our recipe.

## The improvement

Amsel, Persson, Musco & Gower (2025), *"The Polar Express: Optimal Matrix Sign Methods and Their Application to the Muon Algorithm"*, solve for a **different** polynomial at each iteration (minimax-optimal for that step), instead of reusing one. For the same number of steps, the output lands closer to orthogonal; equivalently, the same accuracy takes fewer steps. They report better validation loss than the fixed coefficients on GPT-2-scale training.

The change is cheap here: **optax 0.2.6 already accepts `ns_coeffs` as a tuple of per-step tuples** (`ns_steps` is then ignored), so this is a knob plus a coefficient table, not new Muon code. Take the coefficients from the paper or its reference implementation; do not re-derive or transcribe them from memory.

## Expected payoff: small, and say so up front

- Newton-Schulz costs about 3% of step time at 9 layers (#26 stage 2), so "fewer steps at the same accuracy" saves at most a few percent of wall-clock.
- The quality effect is a second-order refinement of an optimizer that already needs half AdamW's tokens. A tokens-to-CE pair with `min_delta` 6.7M may not be able to see it.

That is why this is worth a cheap CPU check before any card time.

## What to run

1. **CPU (do first, decides whether the GPU half is worth it).** Add `MUON_NS_COEFFS = keller | polar_express` (default `keller`, so the golden run and #26's arms stay bit-identical; test that). Then measure orthogonalization error directly. Take real momentum matrices from a checkpoint's optimizer state, or failing that random matrices of the model's actual shapes (960×960 attention; 960×hidden MLP). Report the singular-value spread after 5 steps under each table, in f32 (Newton-Schulz runs in f32 here, per `_muon`). If Polar Express isn't clearly closer to 1, close as `negative-result` with the numbers.
2. **GPU, only if 1 shows a clear gap.** The #26 stage-2 apparatus: `experiments/recipe/tokens_to_ce.py`, Muon x100, 512 opt steps, 3 seeds, matched, only `MUON_NS_COEFFS` differing. Pre-register in `experiments/recipe/specs/`. **Registered prediction: INCONCLUSIVE.** The effect is plausibly real but below the pair's resolution at 512 steps. A ≥2σ loss kills it. A second arm at fewer steps (e.g. 3 Polar Express steps vs 5 fixed) tests the wall-clock claim instead.

## Novelty

Probably **settled-adjacent**: the method is published and demonstrated at GPT-2 scale. Whether it holds with f16 compute and a Muon/AdamW split is untested here. If the pair shows anything, state the novelty call in the PR per rule 5.

Related: #26 (Muon), #287 (gates this), #360 (weight decay, the higher-leverage Muon recipe fix; run that first).

**Comment by MatthewLacerda2** [comment]
**CPU check (step 1): Polar Express is clearly closer to orthogonal. Proceeding to the pair (PR incoming).**

This is 5 Newton-Schulz steps in numpy f32 with optax's normalization (Frobenius norm + eps, and the transpose for tall matrices). The coefficients are generated by the reference implementation's own code (NoahAmsel/PolarExpress @ 71cc379, `coeffs_list[:5]`), not typed in. The matrices are random with a given singular spectrum at the model's shapes. Real momentum matrices were not used, because loading a checkpoint's optimizer state means JAX next to the running pair.

```
coeffs polar: [(8.237312490495555, -23.157747414558187, 16.6805684114459), (4.082441999064831, -2.8930477353325843, 0.5252849256975642), (3.926347992254648, -2.854746803476524, 0.531802242289498), (3.298218713308519, -2.42454198102671, 0.48632008358844137), (2.297036943455259, -1.636625581259032, 0.4002628455953629)]
(960, 960) gaussian  | keller: mean|s-1|=0.166 p1/p10/p50/p90/p99=0.256/0.701/0.889/1.125/1.198 max=1.202 | polar: mean|s-1|=0.090 p1/p10/p50/p90/p99=0.511/0.872/0.985/1.119/1.131 max=1.131
(960, 960) power -1  | keller: mean|s-1|=0.312 p1/p10/p50/p90/p99=0.390/0.427/0.700/1.135/1.202 max=1.202 | polar: mean|s-1|=0.102 p1/p10/p50/p90/p99=0.743/0.800/0.973/1.120/1.131 max=1.131
(960, 960) power -2  | keller: mean|s-1|=0.960 p1/p10/p50/p90/p99=0.001/0.001/0.002/0.050/0.991 max=1.201 | polar: mean|s-1|=0.940 p1/p10/p50/p90/p99=0.001/0.001/0.004/0.102/1.054 max=1.130
(960, 2560) gaussian  | keller: mean|s-1|=0.143 p1/p10/p50/p90/p99=0.682/0.723/0.930/1.126/1.134 max=1.134 | polar: mean|s-1|=0.085 p1/p10/p50/p90/p99=0.869/0.872/0.965/1.117/1.131 max=1.131
(960, 2560) power -1  | keller: mean|s-1|=0.312 p1/p10/p50/p90/p99=0.390/0.427/0.700/1.135/1.202 max=1.202 | polar: mean|s-1|=0.102 p1/p10/p50/p90/p99=0.743/0.800/0.973/1.120/1.131 max=1.131
(960, 2560) power -2  | keller: mean|s-1|=0.960 p1/p10/p50/p90/p99=0.001/0.001/0.002/0.050/0.991 max=1.201 | polar: mean|s-1|=0.940 p1/p10/p50/p90/p99=0.001/0.001/0.004/0.102/1.054 max=1.130
(2560, 960) gaussian  | keller: mean|s-1|=0.143 p1/p10/p50/p90/p99=0.682/0.723/0.930/1.126/1.134 max=1.134 | polar: mean|s-1|=0.085 p1/p10/p50/p90/p99=0.869/0.872/0.965/1.117/1.131 max=1.131
(2560, 960) power -1  | keller: mean|s-1|=0.312 p1/p10/p50/p90/p99=0.390/0.427/0.700/1.135/1.202 max=1.202 | polar: mean|s-1|=0.102 p1/p10/p50/p90/p99=0.743/0.800/0.973/1.120/1.131 max=1.131
(2560, 960) power -2  | keller: mean|s-1|=0.960 p1/p10/p50/p90/p99=0.001/0.001/0.002/0.050/0.991 max=1.201 | polar: mean|s-1|=0.940 p1/p10/p50/p90/p99=0.001/0.001/0.004/0.102/1.054 max=1.130
```

In summary, mean |σ−1| is **0.085–0.102 (Polar) vs 0.143–0.312 (Keller)** on gaussian and 1/i spectra, and Polar's max is 1.13 against 1.20. On a steep 1/i² spectrum neither table orthogonalizes the tail: 5 steps cannot lift σ≈1e-3 whichever table runs them. So which of these our momentum looks like is what decides the size of the effect, and the pair measures that.
